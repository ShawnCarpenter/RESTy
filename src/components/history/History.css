.History {
  max-width:20vw;
  li {
    list-style-type: none;
  }
}
