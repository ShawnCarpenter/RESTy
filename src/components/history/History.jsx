import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './History.css';

const History = ({ history, onClick }) => {
  const items = history.map((item, index) => (
    <li key={item.url + item.method + index} >
      <HistoryItem
        method={item.method}
        url={item.url}
        onClick={onClick}/>
    </li>));
  return (
    <div className={styles.History}>
      <h3>History</h3>
      <ul>
        {items}
      </ul>
    </div>
  );
};

History.propTypes = {
  history:PropTypes.arrayOf(PropTypes.shape({
    method:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
  }))
};

export default History;
