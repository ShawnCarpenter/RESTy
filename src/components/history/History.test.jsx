import React from 'react';
import { render, cleanup } from '@testing-library/react';
import History from './History';

describe('History component', () => {
  afterEach(() => cleanup());
  it('renders History', () => {
    const { asFragment } = render(<History history ={[
      { method:'POST', 
        url:'http://www.test.api' }
    ]}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
