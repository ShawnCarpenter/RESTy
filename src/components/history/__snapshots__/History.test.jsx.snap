// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<DocumentFragment>
  <div
    class="History"
  >
    <h3>
      History
    </h3>
    <ul>
      <li>
        <div>
          <div>
            POST
          </div>
          <div>
            http://www.test.api
          </div>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
